diff --git a/node_modules/rbxts-transformer-runtime-imports/out/transformer.js b/node_modules/rbxts-transformer-runtime-imports/out/transformer.js
index f04322f..0bd12ac 100644
--- a/node_modules/rbxts-transformer-runtime-imports/out/transformer.js
+++ b/node_modules/rbxts-transformer-runtime-imports/out/transformer.js
@@ -57,7 +57,11 @@ var TransformContext = /** @class */ (function () {
 }());
 exports.TransformContext = TransformContext;
 function cleanImportPath(importPath) {
-    importPath = "_".concat(importPath, "_import_data");
+    // match behavior with roblox-ts in var naming of import data where possible
+	const lastElement = importPath.split("/").pop() ?? importPath;
+
+	// *but* add a second underscore to prevent naming collisions
+	importPath = `__${lastElement}`
     importPath = importPath.replace(/\//gi, "_");
     importPath = importPath.replace(/[^a-z_$]/gi, "");
     return importPath;
@@ -66,19 +70,20 @@ function runServiceCall(methodName) {
     return typescript_1.factory.createCallExpression(typescript_1.factory.createPropertyAccessExpression(typescript_1.factory.createCallExpression(typescript_1.factory.createPropertyAccessExpression(typescript_1.factory.createIdentifier("game"), "GetService"), undefined, [typescript_1.factory.createStringLiteral("RunService")]), methodName), undefined, []);
 }
 function assignROIExpression(importPath) {
-    return typescript_1.factory.createExpressionStatement(typescript_1.factory.createCallExpression(typescript_1.factory.createParenthesizedExpression(typescript_1.factory.createArrowFunction([
-        typescript_1.factory.createModifier(typescript_1.SyntaxKind.AsyncKeyword)
-    ], undefined, [], undefined, undefined, typescript_1.factory.createBlock([
-        typescript_1.factory.createExpressionStatement(typescript_1.factory.createBinaryExpression(typescript_1.factory.createIdentifier(cleanImportPath(importPath)), typescript_1.factory.createToken(typescript_1.SyntaxKind.EqualsToken), typescript_1.factory.createAwaitExpression(typescript_1.factory.createCallExpression(typescript_1.factory.createIdentifier("import"), // how do i put a SyntaxKind.ImportKeyword token in here
-        undefined, [
-            typescript_1.factory.createStringLiteral(importPath)
-        ]))))
-    ]))), undefined, undefined));
-}
-function waitForROILoop(importPath) {
-    return typescript_1.factory.createWhileStatement(typescript_1.factory.createBinaryExpression(typescript_1.factory.createIdentifier(cleanImportPath(importPath)), typescript_1.factory.createToken(typescript_1.SyntaxKind.EqualsEqualsEqualsToken), typescript_1.factory.createIdentifier("undefined")), typescript_1.factory.createBlock([
-        typescript_1.factory.createExpressionStatement(typescript_1.factory.createCallExpression(typescript_1.factory.createPropertyAccessExpression(typescript_1.factory.createIdentifier("task"), "wait"), undefined, undefined))
-    ]));
+    return typescript_1.factory.createVariableStatement(undefined, typescript_1.factory.createVariableDeclarationList([
+		typescript_1.factory.createVariableDeclaration(cleanImportPath(importPath), undefined, undefined, typescript_1.factory.createCallExpression(
+			typescript_1.factory.createPropertyAccessExpression(
+				typescript_1.factory.createCallExpression(
+					typescript_1.factory.createToken(typescript_1.SyntaxKind.ImportKeyword),
+				undefined,
+				[typescript_1.factory.createStringLiteral(importPath)]
+			  ),
+			  typescript_1.factory.createIdentifier("expect")
+			),
+			undefined,
+			[]
+		  )),
+	], typescript_1.NodeFlags.Const));
 }
 function assignDefaultExport(importPath, name) {
     return typescript_1.factory.createExpressionStatement(typescript_1.factory.createBinaryExpression(typescript_1.factory.createIdentifier(name), typescript_1.factory.createToken(typescript_1.SyntaxKind.EqualsToken), typescript_1.factory.createAsExpression(typescript_1.factory.createAsExpression(typescript_1.factory.createPropertyAccessExpression(typescript_1.factory.createIdentifier(cleanImportPath(importPath)), "default"), typescript_1.factory.createTypeReferenceNode("unknown")), getImportType(importPath))));
@@ -92,14 +97,13 @@ function assignNamedBindings(namedBindings, importPath) {
 function runtimeImportBody(importPath, namedBindings, name) {
     return typescript_1.factory.createBlock(__spreadArray([
         assignROIExpression(importPath),
-        waitForROILoop(importPath),
         name ? assignDefaultExport(importPath, name) : undefined
     ], (namedBindings ? assignNamedBindings(namedBindings, importPath) : []), true).filter(function (e) { return e !== undefined; }));
 }
 function getImportType(importPath, qualifier, isTypeOf) {
     if (isTypeOf === void 0) { isTypeOf = true; }
     var importTypeNode = typescript_1.factory.createImportTypeNode(// The overload that's recommended literally doesn't work
-    typescript_1.factory.createLiteralTypeNode(typescript_1.factory.createStringLiteral(importPath)), qualifier, undefined, isTypeOf);
+    typescript_1.factory.createLiteralTypeNode(typescript_1.factory.createStringLiteral(importPath)), undefined, qualifier, undefined, isTypeOf);
     return importTypeNode;
 }
 function visitImportDeclaration(context, node) {
@@ -115,20 +119,19 @@ function visitImportDeclaration(context, node) {
     var namedBindings = clause.namedBindings;
     if (namedBindings && !typescript_1.default.isNamedImports(namedBindings))
         return node;
+	if (!sourceFile)
+		return node;
     var leadingTriviaRange = ((_a = typescript_1.default.getLeadingCommentRanges(sourceFile.getFullText(), node.getFullStart())) !== null && _a !== void 0 ? _a : [])[0];
     if (!leadingTriviaRange)
         return node;
     var leadingTriviaText = sourceFile.getFullText().slice(leadingTriviaRange.pos, leadingTriviaRange.end).toLowerCase();
-    if (!leadingTriviaText.startsWith("//@runtime"))
-        return node;
-    var leadingTriviaSections = leadingTriviaText.split(" ");
-    var leadingTriviaRuntimeArg = leadingTriviaSections[1];
+    const leadingTriviaTextConfig = context.config.leadingText ?? "//@runtime";
+	if(!leadingTriviaText.startsWith(leadingTriviaTextConfig)) return node
+
+	const leadingTriviaRuntimeArg = leadingTriviaText.substring(leadingTriviaTextConfig.length).trim()
     if (leadingTriviaRuntimeArg !== "server" && leadingTriviaRuntimeArg !== "client")
         return node;
     return __spreadArray(__spreadArray([
-        factory.createVariableStatement(undefined, factory.createVariableDeclarationList([
-            factory.createVariableDeclaration(cleanImportPath(path.text), factory.createToken(typescript_1.SyntaxKind.ExclamationToken), getImportType(path.text)),
-        ], typescript_1.NodeFlags.Let)),
         clause.name ?
             factory.createVariableStatement(undefined, factory.createVariableDeclarationList([
                 factory.createVariableDeclaration(clause.name, factory.createToken(typescript_1.SyntaxKind.ExclamationToken), getImportType(path.text)),
